src/Relude/Extra/Foldable1.hs

```diff
@@ -107,6 +107,22 @@ class Foldable f => Foldable1 f where
     minimum1 :: Ord a => f a -> a
     minimum1 = SG.getMin #. foldMap1 SG.Min
 
+    {- | The largest element of a non-empty data structure
+         with respect to the given comparison function.
+
+    >>> maximumOn1 sin (32 :| [64, 8, 128, 16])
+    8.0
+    -}
+    maximumOn1 :: Ord b => (a -> b) -> f a -> a
```

Comment:
If there's no `default` implementation, those functions should be listed in the `{-# MINIMAL #-}` pragma. But I hoped that it should be possible to create a default implementation using `foldr1` or something like that.

Reply:
It should be possible to provide a default implementation if `Foldable1` had a `foldr1`. I can try opening a PR adding `foldr1` to `Foldable1`.

Reply:
For now, you can implement those functions using `toNonEmpty`